spring-kafka/src/main/java/org/springframework/kafka/listener/DefaultAfterRollbackProcessor.java

```diff
@@ -43,19 +45,50 @@
 
 	private static final Log logger = LogFactory.getLog(DefaultAfterRollbackProcessor.class);
 
+	private final FailedRecordTracker failureTracker;
+
+	/**
+	 * Construct an instance with the default recoverer which simply logs the record after
+	 * {@value SeekUtils#DEFAULT_MAX_FAILURES} (maxFailures) have occurred for a
+	 * topic/partition/offset.
+	 * @since 2.2
+	 */
+	public DefaultAfterRollbackProcessor() {
+		this(null, SeekUtils.DEFAULT_MAX_FAILURES);
+	}
+
+	/**
+	 * Construct an instance with the provided recoverer which will be called after
+	 * {@value SeekUtils#DEFAULT_MAX_FAILURES} (maxFailures) have occurred for a
+	 * topic/partition/offset.
+	 * @param recoverer the recoverer.
+	 * @since 2.2
+	 */
+	public DefaultAfterRollbackProcessor(BiConsumer<ConsumerRecord<?, ?>, Exception> recoverer) {
+		this(recoverer, SeekUtils.DEFAULT_MAX_FAILURES);
+	}
+
+	/**
+	 * Construct an instance with the provided recoverer which will be called after
+	 * maxFailures have occurred for a topic/partition/offset.
+	 * @param recoverer the recoverer; if null, the default (logging) recoverer is used.
+	 * @param maxFailures the maxFailures.
+	 * @since 2.2
+	 */
+	public DefaultAfterRollbackProcessor(@Nullable BiConsumer<ConsumerRecord<?, ?>, Exception> recoverer, int maxFailures) {
+		this.failureTracker = new FailedRecordTracker(recoverer, maxFailures, logger);
+	}
+
+	@Override
+	public void process(List<ConsumerRecord<K, V>> records, Consumer<K, V> consumer, Exception exception,
+			boolean recoverable) {
+		SeekUtils.doSeeks(records.stream().collect(Collectors.toList()),
```

Comment:
Well, if you prefer this, let it be. I other place I do something like "double cast" trick:
```
(List<?, ?>)(List) records
```

😄

Reply:
Ha - that works - but STS removes the second cast for me on save. Pushed.